### ISSUE TITLE
Indirect modification of overloaded property Model_Nestedset::$children has no effect

### ISSUE THREAD

**Issue by sagikazarmark** [fuel/orm#300]
This error is thrown, when I try to call dump_tree function with as_object set to true.

In this model EAV is set. The problem is that the eav getter does not return by reference, but by value, so the property does not exists. (UPDATE: the getter DOES return by reference, but it is not a reference of the object in case of EAV.)

**Comment by sagikazarmark** [comment]
Possible solution: return _get_eav by reference. I don't know if that has any effect on other things, but solves this one.

**Comment by sagikazarmark** [comment]
No, it is only a partial solution as this way an error will be thrown, that the property does not exists.

**Comment by sagikazarmark** [comment]
I am not sure, I understand what I did, but I am sure that this problem is related to the fact, that _get_eav does not return by reference.

My PHP version is 5.4.9.

Actually, EAV is not working with Nestedsets at all, but also throws this error. Is this the expected functionality? (I mean, is EAV with Nestedsets implemented? should it work?)

**Comment by sagikazarmark** [comment]
Possible solution: $children and the $path values from dump_tree could be stored in an "exclusion" array, so that properties are always returned by reference.

**Comment by sagikazarmark** [comment]
@WanWizard can you help me understand this issue? I am a bit confused now. :smile:

**Comment by WanWizard** [comment]
Is this still an issue? 

As far as I can see, there is nothing in dump_tree() that could interfere with EAV, nothing accesses properties in that method, only Model object references are assigned to a tree structure.

**Comment by sagikazarmark** [comment]
I will check it out.

**Comment by WanWizard** [comment]
Any results from this checking?
